pkg/reconciler/ingress/resources/virtual_service.go

```diff
@@ -122,6 +122,19 @@ func MakeVirtualServices(ia v1alpha1.IngressAccessor, gateways map[v1alpha1.Ingr
 		requiredGatewayCount += gateways[v1alpha1.IngressVisibilityClusterLocal].Len()
 	}
 
+	// CUSTOM LOGIC TO SUPPORT CUSTOM GATEWAYS PER SERVICE.
+	// This is until Knative supports splitting gateways per service as tracked by this PR (https://github.com/knative/serving/pull/4909).
+	customGateway := ia.GetAnnotations()[networking.CustomGatewayAnnotationKey]
```

Comment:
add a test?

Reply:
So it turns out, the unit test for the component that I am adding into is not on release v.0.9 but in their master. I've tried cherry-picking the commit that added the unit test but it fails as it seems like it is dependent on some previous commits.

Reply:
I'd propose to have 2 PRs, one to our release branch without the unit test, and one to the master branch with the unit tests so that we can push that upstream to Knative. This PR will be the one without the unit test.